cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,179 @@
+
+        serializer_1 = MovieListSerializer(movie_1)
+        serializer_2 = MovieListSerializer(movie_2)
+        serializer_3 = MovieListSerializer(movie_3)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertNotIn(serializer_1.data, response.data)
+        self.assertIn(serializer_2.data, response.data)
+        self.assertNotIn(serializer_3.data, response.data)
+
+    def test_filter_movies_by_genre(self):
+        genre_1 = sample_genre(name="Action")
+        genre_2 = sample_genre(name="Horror")
+
+        movie = sample_movie()
+        movie_Action = sample_movie()
+        movie_Action_Horror = sample_movie()
+
+        movie_Action.genres.add(genre_1)
+        movie_Action_Horror.genres.add(genre_1, genre_2)
+
+        response = self.client.get(
+            MOVIE_URL,
+            {"genres": f"{genre_1.id},{genre_2.id}"}
+        )
+
+        serializer = MovieListSerializer(movie)
+        serializer_Action = MovieListSerializer(movie_Action)
+        serializer_Action_Horror = MovieListSerializer(movie_Action_Horror)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer_Action.data, response.data)
+        self.assertIn(serializer_Action_Horror.data, response.data)
+        self.assertNotIn(serializer.data, response.data)
+
+    def test_filter_movies_by_actor(self):
+        actor_1 = sample_actor()
+        actor_2 = sample_actor()
+
+        movie = sample_movie()
+        movie_actor_1 = sample_movie()
+        movie_actor_2 = sample_movie()
+
+        movie_actor_1.actors.add(actor_1)
+        movie_actor_2.actors.add(actor_2)
+
+        response = self.client.get(
+            MOVIE_URL,
+            {"actors": f"{actor_1.id},{actor_2.id}"}
+        )
+
+        serializer = MovieListSerializer(movie)
+        serializer_actor_1 = MovieListSerializer(movie_actor_1)
+        serializer_actor_2 = MovieListSerializer(movie_actor_2)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer_actor_1.data, response.data)
+        self.assertIn(serializer_actor_2.data, response.data)
+        self.assertNotIn(serializer.data, response.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.actors.add(sample_actor())
+        movie.genres.add(sample_genre())
+
+        url = detail_url(movie.id)
+
+        response = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "TestTitle",
+            "description": "TestDescription",
+            "duration": 150,
+        }
+        response = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
This test case is not clear. It's better to give a meaningful name to the test case that describes what it is testing. For example, it could be `test_non_admin_user_cannot_create_movie` or something similar.